### PR TITLE
obsolete xmlSerialization methods

### PR DESCRIPTION
**Pull request by mjkkirschner**
### Purpose

https://jira.autodesk.com/browse/QNTM-3090

This PR makes the simplest change to the xml serialization methods that I think will help our users and clarify when to use different serialization methods in Dynamo 2.x.

~~This PR simply renames all serialization methods on the modelBase types to use the prefix `Runtime` to indicate that these methods are to be used only for serialization related to copy/paste or undo/redo which are runtime functionalities, not file save or load time.~~

I also added this message to the abstract methods in the modelBase class
``` 
"Use this method only for saving/loading of node state during undo/redo and copy/paste 
- data saved here will not be saved to the .dyn file. This method will be removed in the future to use json"
 ```

Here are other potential fixes I investigated while attempting a refactor and why I choose this approach:

1 - **Full Refactor** 
```
- undo redo uses json serialization which is to be done by viewModel and Model serializers - 
Deserialization of viewModels needs to be refactored again to be useful generally.

It’s unclear how to use the deserializers when we want to reload an element without creating a 
new one… `JsonNet.Populate` should help here but it still seems like it will take some work.

 All model types should have extension methods for `FromJSON` and `ToJSON` ,which can 
replace the `nodeGraph` methods for loading types from XML. Also - it is unclear how a user will 
introduce custom converters for properties that need them. Since we deserialize with specific 
settings I’m not sure if there converters will get picked up… 

**Lastly, after all this work the XML deserialize methods need to remain so that xml loading of
 graphs still works.... I do not think this is the right move right now.*
```

2 - **refactor deserialize and serialize Methods to use JSON**, 
```
just modify the deserialize and serialize methods to pass `JObjects` instead of `xml `objects, this 
means the api is relatively unchanged but we get rid of xml serialization logic, and use the same 
serialization logic 

- We cannot get rid of deserialization logic... :(. Also we rename them to include “Runtime” so its 
clear these methods do not save to the file and instead only save for undo/redo and copy/paste.


Another wrinkle…. as stated above the deserialize methods need work with either `xml` or 
`JObjects` since we are allowing continued opening of xml graphs for some time…. Again, I think
 this is a bunch of work for little gain since we need to keep xml working for deserialize and 
replace all serialization with JSON - the api is still in a strange state afterward and risks are high.

```

3 - **rename so the use is clear**, 
```
create a tutorial documenting the difference between json serialization and xml serialization - and 
when xml serialization is actually required for user devs. The methods are also obsoleted as when we remove the need to open XML refactoring becomes much more useful. - They remain obsolete
 until that happens and then are removed - in place of 1 or 2.
```

#### Other required work if we do this:
~~perform rename in DynamoRevit
perform rename in Reach (do we care?)
perform rename in samples
perform rename in MeshToolkit (unsure)
any other library that includes nodeModels (@saintentropy)~~

### Declarations

Check these if you believe they are true

- [ ] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@QilongTang 
@ramramps 
### FYIs

@saintentropy

### CODE REVIEW

**Comment by mjkkirschner** [comment]
@alfarok dynamoSamples rename

File: src/DynamoCore/Graph/ModelBase.cs [hunk 1]
```diff
@@ -382,8 +386,13 @@ protected virtual XmlElement CreateElement(XmlDocument xmlDocument, SaveContext
             return element;
         }
 
-        protected abstract void SerializeCore(XmlElement element, SaveContext context);
-        protected abstract void DeserializeCore(XmlElement nodeElement, SaveContext context);
+        [Obsolete(@"Use this method only for saving/loading of node state during undo/redo and copy/paste 
+                  - data saved here will not be saved to the .dyn file. This method will be removed in the future to use json")]
+        protected abstract void RuntimeSerializeCore(XmlElement element, SaveContext context);
```
Comment (QilongTang) [inline src/DynamoCore/Graph/ModelBase.cs]: `.dyn` and `.dyf`

File: src/DynamoCore/Graph/ModelBase.cs [hunk 2]
```diff
@@ -382,8 +386,13 @@ protected virtual XmlElement CreateElement(XmlDocument xmlDocument, SaveContext
...
+
+        [Obsolete(@"Use this method only for saving/loading of node state during undo/redo and copy/paste 
+                  - data saved here will not be saved to the .dyn file. This method will be removed in the future to use json")]
+        protected abstract void RuntimeDeserializeCore(XmlElement nodeElement, SaveContext context);
```
Comment (QilongTang) [inline src/DynamoCore/Graph/ModelBase.cs]: same here

**Comment by QilongTang** [comment]
The `Other required work` section is also required if we merge this PR correct?

**Comment by mjkkirschner** [comment]
@QilongTang like we talked about yep. Where did we land on this? We'll just add the obsolete attribute and write some appropriate documentation?

**Comment by QilongTang** [comment]
@mjkkirschner We will not rename them to minimize potential code changes in other projects and simply obsolete them

**Comment by mjkkirschner** [comment]
@QilongTang @ramramps rename reverted, and comment updated.

**Comment by QilongTang** [comment]
@mjkkirschner :shipit:

**Pull request merged**